src/Fixer/LanguageConstruct/ErrorSuppressionFixer.php

```diff
@@ -0,0 +1,125 @@
+ * @author Jules Pietri <jules@heahprod.com>
+ * @author Kuba Werłos <werlos@gmail.com>
+ */
+final class ErrorSuppressionFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Error control operators should be added to deprecation notices and/or removed from other cases.',
+            [new CodeSample("<?php\ntrigger_error('Warning.', E_USER_DEPRECATED);")],
+            null,
+            'Adding or removing ``@`` might cause changes to code behaviour.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_STRING);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('silence_deprecation_error', 'Whether to ensure deprecation notices are silenced.'))
+                ->setDefault(true)
+                ->setAllowedValues([true, false])
+                ->getOption(),
+            (new FixerOptionBuilder('noise_remaining_usages', 'Whether to remove `@` in remaining usages.'))
+                ->setDefault(false)
+                ->setAllowedValues([true, false])
+                ->getOption(),
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $indexes = [];
+
+        for ($index = $tokens->count() - 1; $index >= 0; --$index) {
+            $token = $tokens[$index];
+
+            if ($token->equals('@') && true === $this->configuration['noise_remaining_usages']) {
+                if (!array_key_exists($index, $indexes)) {
+                    $indexes[$index] = false;
+
+                    continue;
+                }
+            }
+
+            if (!$token->equals([T_STRING, 'trigger_error'], false)) {
+                continue;
+            }
+
+            $start = $index;
+            $prev = $tokens->getPrevMeaningfulToken($start);
+            if ($tokens[$prev]->isGivenKind(T_NS_SEPARATOR)) {
+                $start = $prev;
+                $prev = $tokens->getPrevMeaningfulToken($start);
+            }
+
+            if ($tokens[$prev]->isGivenKind(T_STRING) || $tokens[$prev]->equals('@')) {
+                $indexes[$prev] = null;
+
+                continue;
+            }
+
+            $end = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $tokens->getNextTokenOfKind($index, [T_STRING, '(']));
+            if ($tokens[$tokens->getPrevMeaningfulToken($end)]->equals([T_STRING, 'E_USER_DEPRECATED'])) {
+                $indexes[$start] = (true === $this->configuration['silence_deprecation_error']) ? true : null;
```

Comment:
Ternary operator with composed conditions is hard to read, yoda with moreover.
What does this do?

Could you split it to or simplify? Well, comparison bool to `true` could shortened:

```php
$indexes[$start] = $this->configuration['silence_deprecation_error'] ? true : null;
```